Apps/W1/EDocument/app/src/Processing/EDocumentProcessing.Codeunit.al

```diff
@@ -275,6 +275,21 @@ codeunit 6108 "E-Document Processing"
         exit(GetPostedRecord(EDocument, RelatedRecord));
     end;
 
+    /// <summary>
+    /// Deletes related record if it is not posted.
+    /// </summary>
+    /// <param name="EDocument">E-Document record</param>
+    procedure DeleteRelatedRecord(EDocument: Record "E-Document")
+    var
+        RecRef: RecordRef;
+    begin
+        if this.GetRelatedRecord(EDocument, RecRef) then
+            if RecRef.Number = Database::"Purchase Header" then
```

Comment:
This model is insufficient. 
Assumption should be that any document type can be deleted. 
In case that document is posted it cannot. 
How do you know document is posted? E-Doc state Completed should indicate that.